# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/public/IAPZCTreeManager.h
# Commit: c09a51622e98
# Full Hash: c09a51622e988e81077c75e3aaa5a49df66284b3
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-22 16:44:08
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 737807563dd5 -r c09a51622e98 gfx/layers/apz/public/IAPZCTreeManager.h
--- a/gfx/layers/apz/public/IAPZCTreeManager.h	Thu Mar 21 23:14:26 2019 +0000
+++ b/gfx/layers/apz/public/IAPZCTreeManager.h	Fri Mar 22 03:15:14 2019 +0000
@@ -55,7 +55,7 @@
    * up. |aRect| must be given in CSS pixels, relative to the document.
    * |aFlags| is a combination of the ZoomToRectBehavior enum values.
    */
-  virtual void ZoomToRect(const ScrollableLayerGuid& aGuid,
+  virtual void ZoomToRect(const SLGuidAndRenderRoot& aGuid,
                           const CSSRect& aRect,
                           const uint32_t aFlags = DEFAULT_BEHAVIOR) = 0;
 
@@ -81,7 +81,7 @@
    * should be set to ScrollableLayerGuid::NULL_SCROLL_ID.
    */
   virtual void SetTargetAPZC(uint64_t aInputBlockId,
-                             const nsTArray<ScrollableLayerGuid>& aTargets) = 0;
+                             const nsTArray<SLGuidAndRenderRoot>& aTargets) = 0;
 
   /**
    * Updates any zoom constraints contained in the <meta name="viewport"> tag.
@@ -89,7 +89,7 @@
    * the given |aGuid| are cleared.
    */
   virtual void UpdateZoomConstraints(
-      const ScrollableLayerGuid& aGuid,
+      const SLGuidAndRenderRoot& aGuid,
       const Maybe<ZoomConstraints>& aConstraints) = 0;
 
   virtual void SetDPI(float aDpiValue) = 0;
@@ -106,13 +106,13 @@
   virtual void SetAllowedTouchBehavior(
       uint64_t aInputBlockId, const nsTArray<TouchBehaviorFlags>& aValues) = 0;
 
-  virtual void StartScrollbarDrag(const ScrollableLayerGuid& aGuid,
+  virtual void StartScrollbarDrag(const SLGuidAndRenderRoot& aGuid,
                                   const AsyncDragMetrics& aDragMetrics) = 0;
 
-  virtual bool StartAutoscroll(const ScrollableLayerGuid& aGuid,
+  virtual bool StartAutoscroll(const SLGuidAndRenderRoot& aGuid,
                                const ScreenPoint& aAnchorLocation) = 0;
 
-  virtual void StopAutoscroll(const ScrollableLayerGuid& aGuid) = 0;
+  virtual void StopAutoscroll(const SLGuidAndRenderRoot& aGuid) = 0;
 
   /**
    * Function used to disable LongTap gestures.